File: Network Monitor/Pinger.m
//
//  Pinger.m
//  Network Monitor
//
//  Created by HCorbucc on 10/14/14.
//  Copyright (c) 2014 Hugo Corbucci. All rights reserved.
//

#import "Pinger.h"

@implementation Pinger
- (id)initWatching:(NSString* (^)(void)) ipAddressCallback withTimeout: (NSNumber* (^)(void)) timeoutCallback {
    _ipAddressCallback = ipAddressCallback;
    _timeoutCallback = timeoutCallback;
    return [super init];
}

- (BOOL)ping {
    @try {
        NSPipe* output = [NSPipe pipe];
        NSTask* pingTask = [[NSTask alloc] init];
        [pingTask setLaunchPath: @"/sbin/ping"];
        [pingTask setArguments: @[
                                  [NSString stringWithFormat: @"-t%@", _timeoutCallback()],
                                  @"-q",
                                  @"-o",
                                  [NSString stringWithFormat: @"%@", _ipAddressCallback()]
                                  ]];
        [pingTask setStandardOutput:output];
        [pingTask setStandardError:output];
        [pingTask launch];
        [pingTask waitUntilExit];
        return [pingTask terminationStatus] == 0;
    }
    @catch (NSException *exception) {
        return false;
    }
    @finally {
        
    }
    
}

- (NSString *)description {
    return [NSString stringWithFormat: @"ping -t%@ -q -o %@", _timeoutCallback(), _ipAddressCallback()];
}

@end

File: Network Monitor/Pinger.swift
//
//  Pinger.swift
//  Network Monitor
//
//  Created by HCorbucc on 10/20/14.
//  Copyright (c) 2014 Hugo Corbucci. All rights reserved.
//

import Foundation

class Pinger: NSObject {
    let ipAddress : NSString
    let timeout : Int
    init(ipAddress: NSString, timeout: Int) {
        self.ipAddress = ipAddress
        self.timeout = timeout
    }
    func ping() -> Bool {
        let pingTask = NSTask()
        pingTask.launchPath = "/sbin/ping"
        pingTask.arguments = ["-t", timeout.description, "-o", ipAddress.description]
        pingTask.launch()
        pingTask.waitUntilExit()
        return pingTask.terminationStatus == 0
    }
}